azure-pipelines.yml

```diff
@@ -0,0 +1,94 @@
+pool:
+  name: Hosted VS2017
+
+variables:
+  - name: "Build.ArtifactStagingDirectory"
+    value: "Artifacts/"
+  # The following variables are defined as settable at queue time, so they cannot be included here
+  # Branch.CdfProvider: dnn
+  # Branch.CkEditor: development
+  # CakeTarget: BuildAll
+  # ReleaseMode: Beta
+  # RunTests: True
+
+trigger:
+  batch: true
+  branches:
+    include: '*'

```

Comment:
I thought this was an implied setting, not sure if it is necessary. I usually use includes to define specific branches or patterns I want to build.

Example:
```
trigger:
  branches:
    include:
    - main
    - releases/v*
  tags:
    include:
    - v*
```

Reply:
Yes, that's the default.  It felt weird to add the section and then completely rely on the defaults.  It feels like adding the `branches` section gives a hint to someone who might want to adjust the trigger configuration.  If others feel like it's better with it removed, I'm happy to do that as well.

Reply:
@bdukes 
>  It felt weird to add the section and then completely rely on the defaults

This is a really great point and we should update this to include something useful. I do not think it is a wise idea to have it build all branches. Looking at our current branch structure I see there is the `cake-frosting` branch that you are working on. I don't think that should automatically build every time a new commit is pushed. That should only build once the PR is opened. These are just some of my opinions on the topic and if the core team thinks this is fine, go ahead with the merge.

I am thinking something like this may be more appropriate

```
trigger:
  branches:
    include:
    - develop
  tags:
    include:
    - v*
```

This will build any push to the `develop` branch and any tag that is prefixed with `v`. For example `v10.0.0` will trigger a build.

(see PR comment below for how this will handle PRs)

Reply:
For the trigger on v* tags, I would like to do something later after we move (if we succeed) to cake-frosting. On my own projects I have an automation on creation of a release/x.x.x branch. It produces an RC pre-release with automated release notes and mentions based on merged PRs with matching milestone. Then when merging a release/x.x.x branch into master, it tags and produces a release with the same things but not RC.

Reply:
I like the v* tag because I can tag builds for production release and release client. It is something that you can apply regex to parse the version depending on the defined convention.

For example
* v10.0.0
* v10.0.0-RC1

Both of those tags will trigger a build. I have a build step using a regex that checks for the -RC1 which will then generate the build as a Release Client. If the regex matches the production build format of v10.0.0 it will generate the production build

Reply:
yep, the v tags stay the same with what I have, it's just not the trigger. The triggers are creating a release/1.2.3 branch which fires up a series of events:
- Tags v1.2.3-rc1 (or rc2 or rc3 on next commits to that branch)
- Produces release notes
- Produces an unpublished release with the artifacts and all.

Then when release/1.2.3 get's merged into master, same process other than the tag is v1.2.3 (no rc-*)

It's about the same as what I currently do manually for each release other than not having to do it manually 😄 

On top of that when we get this working we also have the proper triggers to adjust the issue templates because we need 2 events for that, when we release an RC and when we release a non-RC.

But yeah, the tags would be the exact same